tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_type_runtime.bal

```diff
@@ -245,6 +245,50 @@ function baz2() returns never[] {
     return [];
 }
 
+function testNeverRestFieldType() {
```

Comment:
Shall we add more tests with inclusive record type-descriptors and exclusive record type-descriptors with different non-never types?

Reply:
This fix is specific to rest field types. Since, we can't have rest field descriptors insideinclusive record type-descriptors, IMO adding tests for them seems redundant. 
I added tests for exclusive record type-descriptors in latest PR.